### PR TITLE
A4A: Address the Pressable Referrals feedback from the CfT post. Copy tweaks.

### PR DESCRIPTION
**Pull request by cleacos**
Resolves: https://github.com/Automattic/automattic-for-agencies-dev/issues/1523

## Proposed Changes

This PR addresses the feedback from the Pressable Referral CfT post. It updates some copy.

**Agency View:**

image

**Client View:**

image

## Why are these changes being made?

*

## Testing Instructions

- Check the code and the new copy.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/a8c-for-agencies/sections/marketplace/checkout/submit-payment-info.tsx [hunk 1]
```diff
@@ -67,9 +67,14 @@ export default function SubmitPaymentInfo( { disableButton }: { disableButton?:
 		if ( status === 'success' ) {
 			dispatch( recordTracksEvent( 'calypso_a4a_client_checkout_submit_payment_info_success' ) );
 			dispatch(
-				successNotice( translate( 'Thank you for your purchase!' ), {
-					displayOnNextPage: true,
-				} )
+				successNotice(
+					paymentMethodRequired
```
Comment (jkguidaven) [inline client/a8c-for-agencies/sections/marketplace/checkout/submit-payment-info.tsx]: I'm not sure why we need to use the `paymentMethodRequired` flag here. Can we just always use the `Thank you for your purchase! Your agency can now set up your product.` text here?

Comment (cleacos) [reply client/a8c-for-agencies/sections/marketplace/checkout/submit-payment-info.tsx]: Good catch. I was thinking to use isClient and I don't why I used that variable. I just updated it.

**Review by jkguidaven — APPROVED** [review]
Thank you for addressing my comments. LGTM 🚢

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17025834).

**Pull request merged**